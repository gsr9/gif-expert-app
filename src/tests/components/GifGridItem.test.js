import { shallow } from "enzyme";
import React from "react";
import { GifGridItem } from "../../components/GifGridItem";

const title = "Título";
const url = "https://i1.wp.com/aprendergratis.es/wp-content/uploads/2018/11/udemy-cursos-gratis.png?resize=1200%2C630&ssl=1";
const wrapper = shallow(<GifGridItem title={title} url={url} />);

describe("Pruebas en el componente <GifGridItem />", () => {
  test("debe de mostrar correctamente el componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener un párrafo con el title", () => {
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);
  });

  test("debe de tener una imagen con src y title", () => {
    const img = wrapper.find("img");

    const { src, alt } = img.props();

    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe de tener animate__fadeIn", () => {
    const div = wrapper.find("div");

    expect(div.hasClass("animate__fadeInUp")).toBe(true);
  });
});
