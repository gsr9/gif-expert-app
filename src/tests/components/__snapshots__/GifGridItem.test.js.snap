// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem /> debe de mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Título"
    src="https://i1.wp.com/aprendergratis.es/wp-content/uploads/2018/11/udemy-cursos-gratis.png?resize=1200%2C630&ssl=1"
  />
  <p>
     
    Título
     
  </p>
</div>
`;
